include/swift/AST/DiagnosticsParse.def

```diff
@@ -129,12 +129,17 @@ ERROR(lex_invalid_unicode_scalar,none,
 ERROR(lex_unicode_escape_braces,none,
       "expected hexadecimal code in braces after unicode escape", ())
 ERROR(lex_illegal_multiline_string_start,none,
-      "invalid start of multi-line string literal", ())
+      "multi-line string literal content must begin on a new line", ())
 ERROR(lex_illegal_multiline_string_end,none,
-      "invalid end of multi-line string literal", ())
-ERROR(lex_ambiguous_string_indent,none,
-      "invalid mix of multi-line string literal indentation", ())
-
+      "multi-line string literal closing delimiter must begin on a new line", ())
+ERROR(lex_inconsistent_string_indent,none,
+      "unexpected %select{space in|tab in|end of}2 indentation of "
```

Comment:
I like this version quite a bit. My only suggestion here would be: `%select{unexpected space in|unexpected tab in|insufficient}2`. IMO, it's just much more easily interpreted when we call it "insufficient indentation of line" than "unexpected end of indentation of line." Looks good though.